mobile/ios/ios-device.ts

```diff
@@ -320,11 +327,130 @@ export class IOSDevice implements Mobile.IIOSDevice {
+                this.$logger.info("Connected localhost " + localPort + " to device " + devicePort);
+
+				buff.forEach((data) => {
+					deviceSocket.write(data);
+				});
+
+				deviceSocket.on('data', (data: NodeBuffer) => {
+					localSocket.write(data);
+				});
+
+				deviceSocket.on('end', (data: NodeBuffer) => {
+                    this.$logger.info("Device socket closed!");
+					process.exit(0);
+				});
+
+			});
+
+            localServer.listen(localPort, () => {
+                this.$logger.info("Opened localhost " + localPort);
+            });
+        });
+    }
+
+    private printHowToTerminate() {
+        this.$logger.info("\nSetting up debugger proxy...\n\nPress Ctrl + C to terminate!\n");
+    }
+
+    private postDebugNotification(packageName: string, notification: string, notificationProxyClient: iOSProxyServices.NotificationProxyClient): void {
+        var attachRequestMessage = packageName + ":NativeScript.Debug.AttachRequest";
+        notificationProxyClient.postNotification(attachRequestMessage);
+        this.$logger.info("Send " + notification);
+    }
+
+    private attachAtStartup(packageName: string) {
+        this.openSafariFrontEnd(packageName);
+
+        var _injector = this.$injector;
+        var notificationProxyClient: iOSProxyServices.NotificationProxyClient = _injector.resolve(iOSProxyServices.NotificationProxyClient, { device: this });
+
+        this.printHowToTerminate();
+
+        var appLaunchMessage = packageName + ":NativeScript.Debug.AppLaunching";
+        notificationProxyClient.addObserver(appLaunchMessage, () => {
+            this.$logger.info("Got AppLaunching");
+
+            this.attachWhenReady(packageName, notificationProxyClient);
+            this.postDebugNotification(packageName, "WaitForDebugger", notificationProxyClient);
+        });
+    }
+
+    private attachAtRuntime(packageName: string): void {
+        this.openSafariFrontEnd(packageName);
+
+        var _injector = this.$injector;
+        var notificationProxyClient: iOSProxyServices.NotificationProxyClient = _injector.resolve(iOSProxyServices.NotificationProxyClient, { device: this });
+
+        this.printHowToTerminate();
+        this.attachWhenReady(packageName, notificationProxyClient);
+        this.postDebugNotification(packageName, "AttachRequest", notificationProxyClient);
+    }
+
+    private attachDebugger(packageName: string): void {
+        this.attachAtRuntime(packageName);
+    }
+
+	private startAppWithDebugger(packageFile: string, packageName: string): void {
+        this.deploy(packageFile, packageName).wait();
+        this.attachAtStartup(packageName);
+        this.runApplication(packageName).wait();
 	}
 
     public debug(packageFile: string, packageName: string): IFuture<void> {
         return (() => {
-            this.$errors.fail({formatStr:"this will come in a future version", suppressCommandHelp: true});
+            if (options["get-port"]) {
```

Comment:
This line is correct, and I do not insist on changing it. Just noting that you can use ```options.getPort``` instead for a slightly shorter code.